src/docs/autoscaling.md

```diff
@@ -0,0 +1,115 @@
+# GOV.UK Notify AutoScaler
+Notify has bursty, periodic loads and traffic. During the night, the traffic stays low and most of the time, is consist of background scheduled testing jobs; and twe get more traffic during office hours. Once in a while, we would receive a task that is consisted of >20k notifications. This has motivated us to develop an autoscaler to response to these tasks and scale up and down Notify apps instances running on PaaS. 
+
+These terms used in this article refer to:
+
+* a **task** is used to described when a user upload a csv file consist of a batch of notifications to be sent 
+* a **job** is a unit of notification, database query or a process to be run on the servers at a time
+* a **request** describes an 'http request' to retrieve or process information.  
+ 
+## Metrics ##
+Notify's auto scaling uses these metrics:
+
+* **RequestCounts (for Request handlers)** <br>
+A metric returned by AWS Cloudwatch for the Elastic Load Balancers (ELB) to scale each of the ELB apps running on PaaS
+
+* **ApproximateNumberOfMessages (for Delivery workers)** <br>
+A metric returned by AWS Cloudwatch for the SQS queues to scale each notification delivery and database apps on PaaS.
+
+* **Number of scheduled notifications (for Scheduled jobs)**<br>
+A metric obtained from the database for csv files to be sent in the next 1 minutes, and is updated everytime when the autoscaler is scheduled run.
+
+The autoscaler is scheduled to run every 20 seconds using the python ```sched``` module. Autoscaling applies to each individual apps. 
+
+
+## Scaling instances ##
+A factor `desired_instances_count` is used to determine the desired number of instances to run for an App at an evaluated period. This factor is calculated differently for our three types of apps and is bounded by:
+ 
+`min_instance_count<=desired_instances_count<=max_instance_count`
+
+The definition of `min_instance_count` and `max_instance_count` can be found in the latter section. 
+
+### Delivery workers ###
+The autoscaler checks the total number of messages in the queues (ApproximateNumberOfMessages) of the app and scaled accordingly. One app can use multiple queues. This scaling is applicable to database tasks, CSV job processor, notifications senders and various research and priority jobs. 
+
+For example, 
+```notify-delivery-worker-sender``` is an app for delivering the notifications to the service providers, i.e. AWS SES and text messaging providers. The celery worker consumes job from two queues: ```send-sms-tasks``` and ```send-sms-tasks``` are being monitored. The number of instances is then calculated as:-
+
+```
+desired_instances_count = total_message_count / request_per_instance
```

Comment:
I think you mean `desired_instance_count = total_message_count / messages_per_instance`